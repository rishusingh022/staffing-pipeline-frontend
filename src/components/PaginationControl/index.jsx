import React from 'react';
import PropTypes from 'prop-types';
import { PAGE_LIMIT } from '../../constants/page';
import { GrFormNextLink } from 'react-icons/gr';
import { GrFormPreviousLink } from 'react-icons/gr';

function PaginationControl({ pageNumber, setPageNumber, objectCount }) {
  const pageCount = Math.ceil(objectCount / PAGE_LIMIT);
  const inputRef = React.useRef(null);
  const goNext = () => {
    if (pageNumber >= pageCount) return;
    inputRef.current.value = Number(pageNumber) + 1;
    setPageNumber(prev => Number(prev) + 1);
  };
  const goPrev = () => {
    if (pageNumber > 1) {
      inputRef.current.value = Number(pageNumber) - 1;
      setPageNumber(prev => Number(prev) - 1);
    } else {
      setPageNumber(1);
    }
  };

  const handleKeyDown = event => {
    if (event.key === 'Enter') {
      const value = Number(event.target.value);
      handleChange(value);
    }
  };

  const handleChange = value => {
    if (value > pageCount) {
      setPageNumber(pageCount);
      inputRef.current.value = pageCount;
    } else if (value < 1) {
      setPageNumber(1);
      inputRef.current.value = 1;
    } else {
      setPageNumber(value);
      inputRef.current.value = Number(value);
    }
  };

  return (
    <div className="relative min-h-[50px] py-4 flex w-full px-10 justify-center">
      <div className="flex gap-12 items-center justify-center">
        <GrFormPreviousLink className="hover:cursor-pointer" onClick={goPrev} size={24} />
        <input
          className="bg-gray-200 border p-1 w-12 text-center rounded-md text-sm "
          defaultValue={pageNumber ? pageNumber : 1}
          type="number"
          ref={inputRef}
          onKeyDown={handleKeyDown}
          onBlur={e => handleChange(e.target.value)}
        />
        <GrFormNextLink className="hover:cursor-pointer" onClick={goNext} size={24} />
      </div>
    </div>
  );
}

PaginationControl.propTypes = {
  type: PropTypes.string,
  pageNumber: PropTypes.number,
  setPageNumber: PropTypes.func,
  count: PropTypes.number,
  objectCount: PropTypes.number,
  setObjectCount: PropTypes.func,
};

export default PaginationControl;
