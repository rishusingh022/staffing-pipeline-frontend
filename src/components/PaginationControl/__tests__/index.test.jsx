import React from 'react';
import PaginationControl from '../index';
import { render } from '@testing-library/react';
// eslint-disable-next-line no-unused-vars
import { screen, fireEvent } from '@testing-library/react';

describe('Pagination control', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<PaginationControl />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const { asFragment } = render(<PaginationControl pageNumber={1} setPageNumber={1} objectCount={1} />);
    expect(asFragment()).toMatchSnapshot();
  });

  // it('should call goNext function when is next button clicked', () => {
  //   const goNext = jest.fn();
  //   render(<PaginationControl pageNumber={1} objectCount={1} />);
  //   const nextButton = screen.getByTestId('next-button');
  //   fireEvent.click(nextButton);
  //   expect(goNext).toHaveBeenCalled();
  // });

  // it('should call goPrev function when is prev button clicked', () => {
  //   const goPrev = jest.fn();
  //   render(<PaginationControl pageNumber={1} objectCount={1} />);
  //   const prevButton = screen.getByTestId('prev-button');
  //   fireEvent.click(prevButton);
  //   expect(goPrev).toHaveBeenCalled();
  // });

  // it('should call handleChange function when is input value changed', () => {
  //   const handleChange = jest.fn();
  //   render(<PaginationControl pageNumber={1} objectCount={1} />);
  //   const input = screen.getByTestId('input');
  //   fireEvent.change(input);
  //   expect(handleChange).toHaveBeenCalled();
  // });
});
