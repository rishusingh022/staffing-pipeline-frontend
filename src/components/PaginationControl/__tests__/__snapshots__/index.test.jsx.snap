// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination control should render correctly 1`] = `
<DocumentFragment>
  <div
    class="relative min-h-[50px] py-4 flex w-full px-10 justify-center"
  >
    <div
      class="flex gap-12 items-center justify-center"
    >
      <svg
        class="hover:cursor-pointer"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6,12.4 L18,12.4 M12.6,7 L18,12.4 L12.6,17.8"
          fill="none"
          stroke="#000"
          stroke-width="2"
          transform="matrix(-1 0 0 1 24 0)"
        />
      </svg>
      <input
        class="bg-gray-200 border p-1 w-12 text-center rounded-md text-sm "
        type="number"
        value="1"
      />
      <svg
        class="hover:cursor-pointer"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6,12.4 L18,12.4 M12.6,7 L18,12.4 L12.6,17.8"
          fill="none"
          stroke="#000"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Pagination control should render correctly with props 1`] = `
<DocumentFragment>
  <div
    class="relative min-h-[50px] py-4 flex w-full px-10 justify-center"
  >
    <div
      class="flex gap-12 items-center justify-center"
    >
      <svg
        class="hover:cursor-pointer"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6,12.4 L18,12.4 M12.6,7 L18,12.4 L12.6,17.8"
          fill="none"
          stroke="#000"
          stroke-width="2"
          transform="matrix(-1 0 0 1 24 0)"
        />
      </svg>
      <input
        class="bg-gray-200 border p-1 w-12 text-center rounded-md text-sm "
        type="number"
        value="1"
      />
      <svg
        class="hover:cursor-pointer"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6,12.4 L18,12.4 M12.6,7 L18,12.4 L12.6,17.8"
          fill="none"
          stroke="#000"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
